### PR TITLE
Maps

### PR DESCRIPTION
**Pull request by msoucy**
Create a system for dynamically getting a RobotMap via reflection

### CODE REVIEW

File: src/main/java/com/chopshop166/chopshoplib/RobotUtils.java [hunk 1]
```diff
@@ -128,4 +134,94 @@ public static double sps(final double value) {
...
+    public static String getMACAddress() {
+        try {
+            final NetworkInterface iface = NetworkInterface.getByName("eth0");
+            final byte[] mac = iface.getHardwareAddress();
+
+            if (mac == null) { // happens on windows sometimes
+                throw new SocketException();
+            }
+
+            final StringJoiner sb = new StringJoiner("-");
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/RobotUtils.java]: s/-/:/

File: src/main/java/com/chopshop166/chopshoplib/RobotUtils.java [hunk 2]
```diff
@@ -128,4 +134,94 @@ public static double sps(final double value) {
...
+        }
+    }
+
+    /**
+     * Get a RobotMap for the given name.
+     * 
+     * @param rootClass The root class object that the map derives from.
+     * @param pkg       The package to look in.
+     */
+    public static <T> T getRobotMap(final Class<T> rootClass, final String pkg) {
```
Comment (bot190) [inline src/main/java/com/chopshop166/chopshoplib/RobotUtils.java]: Can we make this take `T` and get `T.class` inside this function? I would like to hide as much of the boilerplate we can.

Comment (msoucy) [reply src/main/java/com/chopshop166/chopshoplib/RobotUtils.java]: I wish, but the way Java generics work says no - once you get inside the method you can't get a reference to the class object for a generic parameter

Comment (bot190) [reply src/main/java/com/chopshop166/chopshoplib/RobotUtils.java]: That's unfortunate. Luckily the example you provide is straightforward.

**Pull request merged**